File: Ecoverse-API/models/Item.js
// Ecoverse-API/models/Item.js (Full Code)

const mongoose = require('mongoose');

const ItemSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true
  },
  description: {
    type: String,
    required: true,
  },
  type: { // E.g., 'boost', 'seed', 'potion', 'cosmetic'
    type: String,
    required: true,
  },
  costGP: { // Harga Green Points
    type: Number,
    default: 100,
  },
  effect: { // Deskripsi efek atau buff yang diberikan
    type: String,
    default: 'None',
  },
  icon: { // Path atau ID untuk ikon item di frontend
    type: String,
    default: 'default_item',
  },
}, { timestamps: true });

module.exports = mongoose.model('Item', ItemSchema);

File: eco12/lib/models/item.dart
// lib/models/item.dart (Full Code)

class ShopItem {
  final String id;
  final String name;
  final String description;
  final String type;
  final int costGP;
  final String icon;

  ShopItem({
    required this.id,
    required this.name,
    required this.description,
    required this.type,
    required this.costGP,
    required this.icon,
  });

  factory ShopItem.fromJson(Map<String, dynamic> json) {
    return ShopItem(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      type: json['type'],
      costGP: json['costGP'],
      icon: json['icon'],
    );
  }
}